### ISSUE TITLE
Hyperlinks in "About" doesn't work (AppImage)

### ISSUE THREAD

**Issue by RafaelLinux** [JacobDev1/xl-converter#54]
### Version

1.0.2

### OS

Linux

### Description

Buttons on "About" tab, doesn't work, except "Check version". 

I would like to take this opportunity to congratulate you on this comprehensive programme with a very complete interface. If you allow me to suggest something else here, it would be that the checking of updates and their download proposal should be automatic (the AppImages allow it).

### Steps to Reproduce

1. Go to "About" tab
2. Click on each button in that tab

### Expected Result

To open websites or open a window

### Current Result

Only "Check for updates" do something

### Input Formats

Not related

### Output Format

Not related

### Additional Information (Optional)

I can help in yhr ytsndlsyion to Spanish

**Comment by JacobDev1** [comment]
Hi, thanks for your opinion and suggestions.

What distro and desktop environment are you using?

Try running the application from the terminal and share any logs that appear after pushing those buttons.

**Comment by RafaelLinux** [comment]
I'm sorry didn't include more detailed info about. The form seems to not ask about it.

Operating System: openSUSE Tumbleweed 20240809
KDE Plasma Version: 6.1.4
KDE Frameworks Version: 6.4.0
Qt Version: 6.7.2
Kernel Version: 6.10.3-1-default (64-bit)
Graphics Platform: Wayland

**Comment by JacobDev1** [comment]
I found the problem.

Apparently, KDE Plasma is known to have issues with opening hyperlinks.

Anyone curious can run my program (`v1.0.2`) from the terminal to investigate this issue.

I'll try to find workarounds. Unfortunately, they are unlikely to be entirely reliable. [Python docs](https://docs.python.org/3/library/webbrowser.html) state:
> Some way of reliably detecting KDE would be nice; the KDEDIR variable is not sufficient.

Qt has problems, but Python's `webbrowser` isn't 100% reliable either. It's a big headache.

**Comment by RafaelLinux** [comment]
Curiously, I have looked at another application programmed in Python, and it shows hyperlinks directly, not buttons, so it is very likely that the problem is the errors you mention.

In any case, it is a problem that is not important to solve right now.

Remember that if I can collaborate in the translation (although nowadays DeepL does it very well) to Spanish, I can help.

Until now I was using Converseen, but your application seems to me much more complete and offers a lot of information about each option, and that is very important. 

Also, I thank you for having created an AppImage version and not FlatPak (I wouldn't use it in that case).

**Comment by JacobDev1** [comment]
My code is just one line of calling the standard Qt function.

Qt calls a not installed program and even if it's installed, it conflicts with the OS Qt dependencies. It expects 6.6 and the OS ones are 6.7. The bundling may be a factor. Python's built-in `webbrowser` works better, but fails on live ISO.

This seems like a complex issue and what I said may not be accurate. I need some time to wrap my head around it.

Opening text files needs a separate workaround. This is an edge case, Qt usually works well.

**Comment by JacobDev1** [comment]
Issues with opening links to webpages, files, and folders on KDE Plasma will be fixed in `v1.0.3`.

Below are notes for developers facing similar issues.

This problem occurs when running an external process (`subprocess.run` or `QDesktopServices.openUrl`) on a PyInstaller bundled build. Helpful [docs](https://pyinstaller.org/en/stable/common-issues-and-pitfalls.html#launching-external-programs-from-the-frozen-application).

It occurs particularly on KDE Plasma because it's Qt-based.

Error:

```
kde-open: /home/user/Desktop/application/_internal/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by kde-open)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by kde-open)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6KIOGui.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6KIOCore.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6CoreAddons.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6I18n.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6Service.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6Solid.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6WindowSystem.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6ConfigCore.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6Crash.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libKF6AuthCore.so.6)
kde-open: /home/user/Desktop/application/_internal/libQt6Core.so.6: version `Qt_6.7' not found (required by /usr/lib/libQt6Xml.so.6)
```

For me, those were the problematic `os.environ` variables:

```txt
LD_LIBRARY_PATH
QT_PLUGIN_PATH
QML2_IMPORT_PATH
```

**Comment by RafaelLinux** [comment]
Thanks for sharing the trick with programmers  ;)
